packages/ruleset/src/functions/string-boundary.js

```diff
@@ -4,6 +4,22 @@ module.exports = function(schema, _opts, { path }) {
   return validateSubschemas(schema, path, stringBoundaryErrors);
 };
 
+// Rudimentary debug logging that is useful in debugging this rule.
+const debugEnabled = false;
+function debug(msg) {
+  if (debugEnabled) {
+    console.log(msg);
+  }
+}
+
+// An object holding a list of "format" values to be bypassed when checking
+// for the "pattern", "minLength" and "maxLength" fields of a string property, respectively.
+const bypassFormats = {
+  pattern: ['binary', 'byte', 'date', 'date-time', 'url'],
+  minLength: ['date', 'identifier', 'url'],
+  maxLength: ['date', 'url']
```

Comment:
@padamstx What's the thinking for bypassing this check for `url`?

Reply:
@hudlow If we define a property as holding a URL string, should we really require the pattern/minLength/maxLength properties to be defined for it also?   Seemed like overkill to me :)

Reply:
But where do we define the programmatic constraints for a URL if not with these properties? For example:

- How would a client programmatically know what schemes they might expect (e.g., `http`, `https`, `ftp`, `ssh`)?
- How would a client know how much memory or how big a database column they need to safely store the URL?
- How would a client know if a zero-length string might be allowed?

Reply:
I surrender... I'll remove "url" from the list of formats to bypass.

Reply:
I forgot that the context of this rule was checking specifically _request_ properties. Based on the conversations we had when enhancing the security and robustness aspects of API Handbook guidance, I still believe that all inputs need to have precisely defined constraints. The alternative seems like a poorly defined API (which will have compatibility hazards) and/or one with security hazards built in (lack of coherence checking for inputs).

I believe all these constraints ought to be defined for outputs too, but that's probably a discussion for another day.

Reply:
In a discussion about this issue today, it was decided that for a string/url type property, we'll require the maxLength attribute only.  I'll get this change made in the PR...